app/src/main/java/tech/relaycorp/letro/utils/i18n/Locale.kt

```diff
@@ -0,0 +1,23 @@
+package tech.relaycorp.letro.utils.i18n
+
+import java.util.Locale
+
+fun Locale.normaliseString(): String {
+    val languageCode = language.lowercase()
+    val countryCode = country.lowercase()
+    return if (languageCode.isEmpty()) {
+        ""
+    } else if (countryCode.isEmpty()) {
+        languageCode
+    } else {
+        "$languageCode-$countryCode"
+    }
+}
+
+fun String.parseLocale(): Locale {
```

Comment:
I would create a class LocaleHelper in `tech.relaycorp.letro.utils.locale `package. Because we shouldn't add such domain-specific functions  for such common class as `String`. And it would be good to mark method as throwing exception, because there is several exceptions can be thrown:
```

object LocaleHelper {

    @Throws(IndexOutOfBoundsException::class)
    fun fromString(locale: String): Locale {
        try {
            val localeParts = locale.split("-")
            val languageCode = localeParts[0].lowercase()
            val countryCode = localeParts.getOrNull(1)?.uppercase() ?: ""
            val variantCode = localeParts.getOrNull(2) ?: ""
            return Locale(languageCode, countryCode, variantCode)
        } catch (e: Exception) {
            throw InvalidLocaleStringException(e)
        }
    }

}

class InvalidLocaleStringException(cause: Throwable) : Exception(cause)
```

Reply:
> we shouldn't add such domain-specific functions for such common class as `String`

In hindsight, I think you're right.

> And it would be good to mark method as throwing exception, because there is several exceptions can be thrown

There's no input that can throw an exception, unless I missed a test? See: https://github.com/relaycorp/letro-android/blob/809e2345bc4374895dfdd171b884cd90466d3d38/app/src/test/java/tech/relaycorp/letro/utils/i18n/LocaleTest.kt#L54-L106

Reply:
I didn't see the benefit of moving `parseLocale()` into an object -- at least not at this point when it'd be the only thing in it -- so I'm making it a standalone function.

Reply:
> There's no input that can throw an exception, unless I missed a test? See:

`"randomString".parseLocale()` :)

Reply:
Nope, that won't throw an exception.